cql3/statements/create_index_statement.cc

```diff
@@ -169,7 +171,9 @@ std::vector<::shared_ptr<index_target>> create_index_statement::validate_while_e
 
     if (db.existing_index_names(keyspace()).contains(_index_name)) {
         if (!_if_not_exists) {
-            throw exceptions::invalid_request_exception("Index already exists");
+            auto ex = std::make_exception_ptr(exceptions::invalid_request_exception(format("Index {}.{} already exists", keyspace(), _index_name)));
+            mylogger.warn("{}", ex);
```

Comment:
I don't think we should log when the user makes a mistake and we diagnose it. It's not a warning for the operator, it's a user problem. The user will see it in their shell, or their own logs.

Reply:
I agree, I also don't think this change is in a good direction.

Moreover, why should we do this logging for the "index already exists" and not the hundreds of other CQL errors we generate and do not log? If we believe that user errors should be logged, I think we need to do it across the board and not in some random CQL request that bothered us in one specific case.

Reply:
Moreover, I don't even understand how it would have helped you in https://github.com/scylladb/scylladb/issues/21034 - unless I'm misunderstanding something, it seems that in that issue the test told you up front you were trying to create a keyspace that already existed. You didn't need the log to tell you this... In fact, it's much easier to see it in the application (the test), where you know exactly which keyspace you tried to create, or why.

Maybe it makes sense to add the keyspace/table/view/etc. name to the error message. This I agree with (the application knows which name it tried to create, but sometimes it's nicer to see it spelled out in the error message). But not the logging.